examples/tut-6-rpc/client-py/index.py

```diff
@@ -0,0 +1,130 @@
+import json
+import os
+import random
+import requests
+import socket
+import time
+from concurrent.futures import ALL_COMPLETED, ThreadPoolExecutor, wait
+
+
+# https://gist.github.com/butla/2d9a4c0f35ea47b7452156c96a4e7b12
+def wait_for_port(port, host='localhost', timeout=5.0):
+    """Wait until a port starts accepting TCP connections.
+    Args:
+        port (int): Port number.
+        host (str): Host address on which the port should exist.
+        timeout (float): In seconds. How long to wait before raising errors.
+    Raises:
+        TimeoutError: The port isn't accepting connections after time specified
+        in `timeout`.
+    """
+    start_time = time.perf_counter()
+    while True:
+        try:
+            with socket.create_connection((host, port), timeout=timeout):
+                break
+        except OSError as ex:
+            time.sleep(0.01)
+            if time.perf_counter() - start_time >= timeout:
+                raise TimeoutError('Waited too long for the port {} '
+                                   'on host {} to start accepting connections.'
+                                   .format(port, host)) from ex
+
+
+def mk_rpc(proto='http',
+           host=os.environ['REACH_RPC_SERVER'],
+           port=os.environ['REACH_RPC_PORT']):
+
+    def rpc(m, *args):
+        wait_for_port(port, host)
```

Comment:
I'd put think when `mk_rpc` is called so it only happens once. We'll assume that once it works once, all subsequent failures should be exposed

Reply:
Moved `wait_for_port` in bbeabd7.